### PR TITLE
Add Fuchsia support

### PR DESCRIPTION
**Pull request by abarth**
Use the proper name for `libvulkan.so` on Fuchsia.

### CODE REVIEW

File: ash/src/entry.rs [hunk 1]
```diff
@@ -63,11 +63,11 @@ impl Entry {
 
         #[cfg(all(
             unix,
-            not(any(target_os = "macos", target_os = "ios", target_os = "android"))
+            not(any(target_os = "macos", target_os = "ios", target_os = "android", target = "fuchsia"))
```
Comment (MarijnS95) [inline ash/src/entry.rs:66]:
```suggestion
            not(any(target_os = "macos", target_os = "ios", target_os = "android", target_os = "fuchsia"))
```

?

**Review by MarijnS95 — CHANGES_REQUESTED** [review]

**Comment by MarijnS95** [comment]
https://doc.rust-lang.org/reference/conditional-compilation.html#target_os I'm surprised the official docs don't mention the existence of `fuchsia` :/

**Review by Ralith — APPROVED** [review]
Fuchsia doesn't do the unixy `.so.$MAJOR` thing? Interesting; is this documented anywhere?

**Comment by Ralith** [comment]
Note that if you build ash with `features = ["linked"], default-features = false`, existing versions of ash should Just Work, and be less fragile at runtime besides.

**Comment by abarth** [comment]
> Fuchsia doesn't do the unixy .so.$MAJOR thing?

Yes, we do that for most shared libraries.  I'm not sure why `libvulkan.so` doesn't have a `.$MAJOR` suffix.

> Interesting; is this documented anywhere?

https://fuchsia.dev/fuchsia-src/development/graphics/magma/concepts/vulkan?hl=en#buildtime_dependencies

> Any way we can sensibly cover this in CI?

Not yet.  We're working on that as part of our quest for Tier 1 support in Rust.

> Note that if you build ash with features = ["linked"], default-features = false, existing versions of ash should Just Work, and be less fragile at runtime besides.

Ok, I'll give that a try.

**Comment by Ralith** [comment]
> Yes, we do that for most shared libraries. I'm not sure why libvulkan.so doesn't have a .$MAJOR suffix.

Hmm, are you using the standard Khronos Vulkan loader? This may be a bug in the upstream build system, if so. I'm not entirely clear on the significance of respecting that pattern vs. not, but I figure it's probably worth being consistent with the system at large.

**Comment by MarijnS95** [comment]
@abarth any update on not using/having a `.MAJOR` suffix on Fuchsia?

**Review by MarijnS95 — APPROVED** [review]

**Comment by MarijnS95** [comment]
Looking at the linked Fuchsia docs it's still `libvulkan.so` without `.MAJOR` suffix; I'll merge this in for the `0.38` release.

**Comment by MarijnS95** [comment]
For `cargo:rustc-link-lib=vulkan` under the `linked` feature, this should keep working OOTB :)

**Pull request merged**